extension/src/util/config.ts

```diff
@@ -26,6 +35,72 @@ export function getConfigJavaImportGradleJavaHome(): string | null {
     return vscode.workspace.getConfiguration("java").get<string | null>("import.gradle.java.home", null);
 }
 
+export async function getJavaExecutablePath(): Promise<string | undefined> {
+    const REQUIRED_JDK_VERSION = 17;
+
+    // search from jdt.ls.java.home, java.home
+    const javaHomeGetters = [getJdtlsConfigJavaHome, getConfigJavaHome];
+    let javaHome: string | undefined = undefined;
+    let javaVersion = 0;
+
+    for (const getJavaHome of javaHomeGetters) {
+        javaHome = getJavaHome() || undefined;
+        if (javaHome) {
+            javaVersion = await getMajorVersion(javaHome);
+            // Ensure the Java version is greater than 17
+            if (javaVersion >= REQUIRED_JDK_VERSION) {
+                const javaExecPath = path.join(javaHome, "bin", JAVA_FILENAME);
+                if (fs.existsSync(javaExecPath)) {
+                    return javaExecPath;
+                }
+            }
+        }
+    }
+
+    // search valid JDKs from env.JAVA_HOME, env.PATH, SDKMAN, jEnv, jabba, Common directories
+    const javaRuntimes = await listJdks();
+    sortJdksByVersion(javaRuntimes);
+
+    const validJdks = javaRuntimes.filter((r) => r.version!.major >= REQUIRED_JDK_VERSION);
+    if (validJdks.length > 0) {
+        sortJdksBySource(validJdks);
+        javaHome = validJdks[0].homedir;
+        javaVersion = validJdks[0].version!.major;
+    }
```

Comment:
This part follow https://github.com/redhat-developer/vscode-java/blob/38c8b582b40db9644696924f899c73a3251563f4/src/requirements.ts#L74-L100.
Since all this function happens after user already got `redHat.java` install.